### ISSUE TITLE
Request to allow users to use own styles

### ISSUE THREAD

**Issue by modulexcite** [janraasch/tab-ahead#71]
Thank you for the extension, great work.

Would be able to modify the extension for users to be able to add their own styles for the interface? I've modified the default stylesheet of your extension but I can't replace it with my own styles as Google Chrome disables any extension that's modified. The only option is to either load the extension in developer mode but that means I've to deal with the annoying popup from Chrome every time I launch it or publish the extension with my modifications on the Chrome Web Store.

**Comment by janraasch** [comment]
Hi @modulexcite. That is quiet a feature request you have there :)

Let us first try to solve the problem w/o adding a new feature to the extension.

Off the top of my head I would ask you to try two things:

First, have you tried using any of the extensions to add custom styles to Chrome, like https://chrome.google.com/webstore/detail/stylebot/oiaejidbmkiecgbjeifoejpgmdaleoha, https://chrome.google.com/webstore/detail/styler/bogdgcfoocbajfkjjolkmcdcnnellpkb or https://chrome.google.com/webstore/detail/stylish/fjnbnpbmkenffdnngjfgmeleoegfcffe? Maybe you could use one of those as they all already provide an interface with a nice syntax highlighting editor to add custom CSS to webpages and after all an extension popup is only a webpage. (In our case hosted at chrome-extension://naoajjeoiblmpegfelhkapanmmaaghmi/popup.html)

Second, I do not fully understand how you modified the existing `.css` file, but if you 
- download the `TabAhead.zip` file from https://github.com/janraasch/tab-ahead/releases, 
- unzip it, 
- adjust `main.css` and 
- zip it again, 
then you should be able to add the extension via `Load unpacked extension...` when `Developer mode` is enabled under `chrome://extensions/`. I do not believe you will be presented with a warning or error, but let me know, if this does not work for you.

Let me know, if one of these options work for you.

**Comment by modulexcite** [comment]
Hi Jan,
I understand it might be too much to ask, sorry I wasn't trying to be difficult perhaps just a little. :) 
Google has changed their policy recently (not sure when exactly) on extensions being modified locally, hence I can't just modify the extension and use it. The extension has to exactly match what's released on the chrome store and the version you've installed. Hence I can't just pack and load the extension like we used to in the past without chrome warning me about it. On the other hand if I just update the style sheet found in the chrome's extensions folder, then chrome won't even load the extension and will warn that the extension was corrupted and prompts to repair by reinstalling the extension from the store.
Thanks for responding and the extension, much appreciated.

___________________________
Thanks.

		_____________________________

**Comment by janraasch** [comment]
Could you please try the two suggestions I provided.

The second one was:

- download the TabAhead.zip file from https://github.com/janraasch/tab-ahead/releases,
- unzip it,
- adjust main.css

then you should be able to add the extension via Load unpacked extension... when Developer mode is enabled under chrome://extensions/. I do not believe you will be presented with a warning or error, but let me know, if this does not work for you.

This will not show any warnings. I do this for development all the time.

**Comment by modulexcite** [comment]
I wasn't probably clear on this earlier but that's what I've been trying to get at, you can no longer just load an unpacked extension in chrome without the annoying popup every time I restart the browser. It's fine if you just want to debug the extension but not if you want to use it permanently. For that you will have to publish it on the chrome web store. 
I'm using 
**Chrome**: Version 52.0.2743.33 beta-m (64-bit)
**OS**: Windows 7 64-bit
I was on the Chrome Stable channel earlier but thought I'd give this a try on the beta version, both produce similar results.

The following are the screenshots of what happens when I either load an unpacked extension or modify it locally.


**Option 1: Loading an unpacked extension**
![2016-06-09_21-21-26](https://cloud.githubusercontent.com/assets/12729391/15951910/7c611472-2e8a-11e6-8c3c-164ee15a692d.png)

**Option 2: Modifying an extension locally**
Chrome will disable the extension an report it as corrupted if you modify the content of a file that belongs to an extension. Sadly, there is no override feature on this it will force you to reinstall the extension. If you find that you can override this setting please let me know.
![2016-06-09_21-26-11](https://cloud.githubusercontent.com/assets/12729391/15951932/9f5490f8-2e8a-11e6-9af3-d9f0b2fa8854.png)


Thank you for your time.

**Comment by janraasch** [comment]
Alright, I see. I was not aware of this. Seems to be a *Windows-thing*, see https://bugs.chromium.org/p/chromium/issues/detail?id=337734.

Apparently you could switch to the beta channel of Chrome to get rid of the warning as explained in [Comment 23](https://bugs.chromium.org/p/chromium/issues/detail?id=337734#c23) on the issue.

If that is not an option for you, please try

> First, have you tried using any of the extensions to add custom styles to Chrome, like https://chrome.google.com/webstore/detail/stylebot/oiaejidbmkiecgbjeifoejpgmdaleoha, https://chrome.google.com/webstore/detail/styler/bogdgcfoocbajfkjjolkmcdcnnellpkb or https://chrome.google.com/webstore/detail/stylish/fjnbnpbmkenffdnngjfgmeleoegfcffe? Maybe you could use one of those as they all already provide an interface with a nice syntax highlighting editor to add custom CSS to webpages and after all an extension popup is only a webpage. (In our case hosted at chrome-extension://naoajjeoiblmpegfelhkapanmmaaghmi/popup.html)

as mentioned above.

**Comment by janraasch** [comment]
Also, I just came across this extension:

https://chrome.google.com/webstore/detail/quick-tabs/jnjfeinjfmenlddahdjdmgpbokiacbbb

Seems like they offer Custom CSS Styling and lots of other features out of the box.

**Comment by modulexcite** [comment]
I apologize for the delayed response. I switched to the beta channel before reporting the issue here which is what I reported [earlier](https://github.com/janraasch/tab-ahead/issues/71#issuecomment-225074675) to be on and if you look at [comments 24 and 25 by the same user](https://bugs.chromium.org/p/chromium/issues/detail?id=337734#c23), chrome dev channel has the same problem on windows so I didn't bother switching to it.
I have used QuickTabs before but I like this one better as it's simple and concise to what I need.
I haven't tried the other extensions that would allow me to add custom styling but I prefer not to add yet another extension just for the sake of it, I'm not trying to be difficult here I'm only blaming Google for this :rage1: I can use this extension as is and pretty much does what I need, the styling is like a "nice to have" feature and it's not something that I absolutely must have it. Thank you for your time.
Anyway, I appreciate all the help. Thank you very much for being so patient with me on this. :+1:
